app/controllers/spree_cm_commissioner/qr_images_controller.rb

```
@@ -1,6 +1,10 @@
 class SpreeCmCommissioner::QrImagesController < ApplicationController
   def show
-    order = Spree::Order.find_by!(token: params[:id])
+    token = params[:id].match(/^R\d{9,}-([A-Za-z0-9_\-]+)$/)&.captures
+
+    return unless token
```

if the token is not valid we can raise not found status code. We need to add a spec for this.

I think it would be better to have a method in the Order to find the order by qr_data. we test this method instead of the controller.